patches/0003-Implement-crypto-internal-backend.patch

```diff
@@ -901,14 +901,49 @@ index 00000000000000..6abb6ff007c99f
 +func SupportsRSAOAEPLabel(label []byte) bool     { return true }
 +func SupportsPKCS1v15Hash(hash crypto.Hash) bool { return true }
 +
-+func NewMD5() hash.Hash        { return cng.NewMD5() }
-+func NewSHA1() hash.Hash       { return cng.NewSHA1() }
-+func NewSHA224() hash.Hash     { panic("cngcrypto: not available") }
-+func NewSHA256() hash.Hash     { return cng.NewSHA256() }
-+func NewSHA384() hash.Hash     { return cng.NewSHA384() }
-+func NewSHA512() hash.Hash     { return cng.NewSHA512() }
-+func NewSHA512_224() hash.Hash { panic("cngcrypto: not available") }
-+func NewSHA512_256() hash.Hash { panic("cngcrypto: not available") }
++// Wrapper types to handle Clone method with panic on error
++type hashWrapper struct {
++    hash.Hash
++    cloner interface{ Clone() (hash.Hash, error) }
++}
++
++func (h *hashWrapper) Clone() (hash.Hash, error) {
++    cloned, err := h.cloner.Clone()
++    if err != nil {
++        panic("cngcrypto: clone failed: " + err.Error())
++    }
++    return wrapHash(cloned), nil
++}
++
++func wrapHash(h hash.Hash) *hashWrapper {
++    cloner, ok := h.(interface{ Clone() (hash.Hash, error) })
++    if !ok {
++        panic("cngcrypto: hash does not implement Clone() (hash.Hash, error)")
```

Comment:
@qmuntal maybe I should remove the conversion in here, because we are panicking in clone method anyways?